code/modules/power/lighting.dm

```diff
@@ -84,27 +84,28 @@
 	text = ""
 	var/on = 0 // 1 if on, 0 if off
 	var/brightness = 1.6 // luminosity when on, also used in power calculation
-	var/light_status = LIGHT_OK	// LIGHT_OK, _EMPTY, _BURNED or _BROKEN
 
 	var/obj/item/light/light_type = /obj/item/light/tube // the type of the inserted light item
 	var/allowed_type = /obj/item/light/tube // the type of allowed light items
-	var/light_name = "light tube"				// the name of the inserted light item
 
-	var/fitting = "tube"
-	var/breakprob = 0	// probability the light burns out
+	var/inserted_lamp = null // Reference for the actual lamp item inside
+	var/obj/item/light/current_lamp = null // For easily accessing inserted_lamp's variables, which we do often enough. Don't desync these two!
 
+	var/fitting = "tube"
 	var/wallmounted = 1
 	var/nostick = 1 //If set to true, overrides the autopositioning.
 	var/candismantle = 1
-	var/rigged = 0				// true if rigged to explode
-	var/mob/rigger = null // mob responsible for the explosion
+
 	power_usage = 0
 	power_channel = LIGHT
 	var/removable_bulb = 1
 	var/datum/light/point/light
 
 	New()
 		..()
+		inserted_lamp = new light_type()
+		current_lamp = inserted_lamp
+		current_lamp.loc = src
```

Comment:
Generally avoid setting loc directly unless you have a very good reason to do that.
```suggestion
		current_lamp.set_loc(src)
```
`set_loc` is preferred because quite often things need to update various stuff when they move.
Alternatively you can just do:
```
		inserted_lamp = new light_type(src)
		current_lamp = inserted_lamp
```
and it'd be even cleaner.